File: Strings/Java/Reversing_the_vowels.java
class Solution
{
    String modify (String s)
    {
        // your code here
        
        Stack<Character> v = new Stack<>();
        char[] str = s.toCharArray();
        s = "";

        for (int i=0; i<str.length; i++) {
            char temp = str[i];
            if (temp=='a' || temp=='e' || temp=='i' || temp=='o' || temp=='u') {
                v.push(temp);
            }
        }
        
        for(int i=0; i<str.length; i++) {
            char temp = str[i];
            if (temp=='a' || temp=='e' || temp=='i' || temp=='o' || temp=='u') {
                str[i] = v.pop();
            }
            s += str[i];
        }
        return s;
    }
}

File: Solutions/Python/Reversing_the_vowels.py
class Solution:
    def modify(self, s):
        # code here
        vowels = ""
        for char in s:
            if char in "aeiouAEIOU":
                vowels += char     
        result_string = ""
        for char in s:
            if char in "aeiouAEIOU":
                result_string += vowels[-1]
                vowels = vowels[:-1]
            else:
                result_string += char
        return result_string
